Podcast/PlayerEpisodeDetailView.swift

```diff
@@ -80,26 +90,59 @@ class PlayerEpisodeDetailView: UIView {
+                make.bottom.equalToSuperview().inset(seeMoreButtonHeight)
+            })
         } else {
-            episodeArtworkImageView.frame.size = artworkSmallDimension
-            episodeArtworkImageView.frame.origin = CGPoint(x: marginSpacing, y: artworkY)
-            episodeTitleLabel.frame.origin = CGPoint(x: episodeArtworkImageView.frame.maxX + marginSpacing, y: episodeArtworkImageView.frame.origin.y + episodeTitleYInset)
-            episodeTitleLabel.frame.size = CGSize(width: frame.size.width - marginSpacing - episodeTitleLabel.frame.origin.x, height: episodeTitleLabelHeight)
-            dateLabel.frame.origin = CGPoint(x: episodeArtworkImageView.frame.maxX + marginSpacing, y: episodeTitleLabel.frame.maxY + dateLabelYSpacing)
+            episodeArtworkImageView.snp.remakeConstraints({ make in
+                make.size.equalTo(artworkSmallDimension)
+                make.leading.equalTo(marginSpacing)
+                make.top.equalToSuperview().offset(artworkY)
+            })
+            
+            episodeTitleLabel.snp.remakeConstraints({ make in
+                make.top.equalTo(episodeArtworkImageView.snp.top)
+                make.leading.equalTo(episodeArtworkImageView.snp.trailing).offset(episodeTitleShowMoreSpacing)
+                make.trailing.equalToSuperview().inset(trailingSpacing)
+                make.height.equalTo(episodeTitleLabelHeight)
+            })
+            
+            dateLabel.snp.remakeConstraints({ make in
+                make.top.equalTo(episodeTitleLabel.snp.bottom).offset(dateLabelShowMoreTopOffset)
+                make.leading.equalTo(episodeArtworkImageView.snp.trailing).offset(episodeTitleShowMoreSpacing)
+                make.trailing.equalToSuperview().inset(trailingSpacing)
+                make.height.equalTo(dateLabelHeight)
+            })
+        }
+            seeMoreButton.snp.remakeConstraints { make in
+            make.trailing.equalTo(descriptionTextView.snp.trailing)
+            make.top.equalTo(descriptionTextView.snp.bottom)
+            make.height.equalTo(seeMoreButtonHeight)
         }
         
-        dateLabel.frame.size = CGSize(width: frame.width - marginSpacing - dateLabel.frame.origin.x, height: dateLabelHeight)
-        descriptionTextView.frame.origin = CGPoint(x: marginSpacing, y: dateLabel.frame.maxY + descriptionTextViewSpacing)
-        descriptionTextView.frame.size = CGSize(width: frame.size.width - 2 * marginSpacing, height: frame.height - descriptionTextView.frame.origin.y - seeMoreButton.frame.height)
         descriptionTextView.isScrollEnabled = !expandedArtwork
-        
-        seeMoreButton.frame.origin = CGPoint(x: descriptionTextView.frame.maxX - seeMoreButton.frame.width, y: descriptionTextView.frame.maxY)
         seeMoreButton.setTitle(expandedArtwork ? "Show More" : "Show Less", for: .normal)
+        layoutIfNeeded()
```

Comment:
do we need this layout? lol. but actually

Reply:
we do, I tried removing it and the animation looked really weird